### PR TITLE
added radial menu, locked tower and fixed popup name bug

### PR DESCRIPTION
**Pull request by Karbb**
I added a radial menu behaviour to the strategic point menu. You can add as many as buttons you want and they will be set in a radial position around the center of the menu.

Moreover I added a locked bool on the tower, that will lock the tower, changing the icon, disabling the button and the possibility to buy a tower from that slot.

Finally, i fixed a little bug on the popup. Now it shows the name of the tower and not the numeric value :D

Before submitting a Pull Request please read and tick this checkbox:

- [X] I did read the [project licensing](https://github.com/crystal-bit/hacktoberfest-2020#license) and I agree with the content of this Pull Request being licensed under the same conditions.

### CODE REVIEW

File: scenes/strategic-point-menu.gd [hunk 1]
```diff
@@ -15,6 +15,10 @@ func _on_Slot_pressed(is_slot_active, slot: Slot):
 		buy_tower(slot)
 
 
+func get_slots():
+	return $RadialContainer.get_children()
```
Comment (davcri) [inline scenes/strategic-point-menu.gd:19]: Can you add `$RadialContainer` as a onready var?

**Review by davcri — CHANGES_REQUESTED** [review]
I requested a small change in the code. Also can you rename `scenes/ui/RadialContainer.gd ` to `scenes/ui/radial-container.gd`?

**Comment by davcri** [comment]
Thanks @Karbb for all the changes! I really like the new radial menu.
 
Also a small issue that I noticed: the name that appears in the popup doesn't seem in sync with the tower data. See attached picture. I tried taking a look at the code but I didn't find the culprit.

image

File: scenes/strategic-point-menu.gd (hunk 1 above)
Comment (Karbb) [reply scenes/strategic-point-menu.gd]: yeppa

**Comment by Karbb** [comment]
Made all modifications requested and fixed some stuff that @davcri addressed during his last livestream.

Changed node type from container to control
Removed logic from process method
ADDED JUICEEE aka animations è.è

**Review by davcri — APPROVED** [review]
Nice!

**Comment by davcri** [comment]
@Karbb the new appear animation is cool :sunglasses: However maybe we will speed it up in the future for a better game feel.

Regarding the naming issue I'l going to open an issue about that.

**Comment by davcri** [comment]
Issue created: https://github.com/crystal-bit/hacktoberfest-2020/issues/68

**Pull request merged**